src/widgets/GitResetToRemoteForm.tsx

```diff
@@ -0,0 +1,68 @@
+import { Dialog } from '@jupyterlab/apputils';
+import { TranslationBundle } from '@jupyterlab/translation';
+import { Widget } from '@lumino/widgets';
+import { Git } from '../tokens';
+
+/**
+ * The UI used for the Reset to Remote Form,
+ * used as the body for Reset to Remote Dialog.
+ */
+export class GitResetToRemoteForm
+  extends Widget
+  implements Dialog.IBodyWidget<Git.IGitResetToRemoteFormValue>
+{
+  constructor(
+    trans: TranslationBundle,
+    promptContent = trans.__(
+      'To bring the current branch to the state of its corresponding remote tracking branch, \
+      a hard reset will be performed, which may result in some files being permanently deleted \
+      and some changes being permanently discarded. Are you sure you want to proceed? \
+      This action cannot be undone.'
+    ),
+    warningCloseAllOpenedFilesContent = trans.__(
+      'Also close all opened files to avoid conflicts'
+    )
```

Comment:
Let's move the arguments definition to the caller - this allows to remove the translation object too.

```suggestion
    textBody: string,
    checkboxLabel: string
```